### DISCUSSION TITLE
Load ajax html when click button

### DISCUSSION THREAD

**Discussion by thanhansoft** [filamentphp/filament#12452] [Help]
### Package

Form builder

### Package Version

v3

### How can we help you?

Hi,
I have difficulty using ajax with filament 3.
I want when I press a button, the html content will load using ajax.
In that html, the select query will select the elements.
I usually do it manually with jquery like this:

`$(".add-row").click(function(){
        $.ajax({ type: "GET",
            url: "'~url.get('admin/care/loadProducts/')~'",
            async: false,
            success : function(response){
                //console.log(response);
                $("#insert-row").append(response);
                $.bindField();
            }
        });
    });`

![123](https://github.com/filamentphp/filament/assets/2970872/82e76606-4381-485b-977e-e60ad8da82eb)

Please suggest me how to do this.

**Comment by leandrocfe** [comment]
You could use a `hintAction` or `suffixAction` to load the data

```php
use Filament\Forms\Components\Select;
use Illuminate\Support\Arr;
use Illuminate\Support\Facades\Http;

Select::make('select')
    ->hintAction(Forms\Components\Actions\Action::make('loadOptions')
        ->action(function (Select $component) {
            $data = Http::get('https://jsonplaceholder.typicode.com/users')->json();
            $data = Arr::pluck($data, 'name', 'id');
            $component->options($data);
        }))
```

**Comment by thanhansoft** [comment]
Thanks for support.
if Filament form can do all the manual requirements, I will research and make cms with filament
